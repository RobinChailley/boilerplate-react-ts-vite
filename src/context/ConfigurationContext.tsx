import { ConfigContextValueType } from '@context/configurationContextTypes';
import getConfigContextValue from '@context/getConfigurationContextValue';
import { createContext, useContext } from 'react';

export const ConfigContext = createContext<ConfigContextValueType>(getConfigContextValue());

export const useConfig = () => useContext(ConfigContext);
