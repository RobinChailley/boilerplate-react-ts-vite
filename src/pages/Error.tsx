import React from 'react';
import { useRouteError } from 'react-router-dom';

interface ErrorProps {

}

const Error: React.FC<ErrorProps> = () => {
    const error = useRouteError();
    console.error(error);

    return (
        <div id="error-page">
            <h1>Oops!</h1>
            <p>Sorry, an unexpected error has occurred.</p>
        </div>
    );
};

export default Error;
