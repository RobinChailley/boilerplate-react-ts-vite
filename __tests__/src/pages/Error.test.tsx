import { ConfigContext } from '@context/ConfigurationContext';
import Error from '@pages/Error';
import renderer from 'react-test-renderer';

import getConfigContextValue from '../../mocks/context/getConfigurationContextValue';

describe('Error', () => {
  it('should render correctly', () => {
    const render = renderer.create(
      <ConfigContext.Provider value={getConfigContextValue()}>
        <Error />
      </ConfigContext.Provider>,
    );

    const snapshot = render!.toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
