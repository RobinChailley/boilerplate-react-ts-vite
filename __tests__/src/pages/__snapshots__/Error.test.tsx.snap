// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should render correctly 1`] = `
<div
  id="error-page"
>
  <h1>
    Oops!
  </h1>
  <p>
    Sorry, an unexpected error has occurred.
  </p>
</div>
`;
